### ISSUE TITLE
set!(u::Field{G}, f::Function) should use xC, yC, zC (or xF, yF, zF) depending on Field.

### ISSUE THREAD

**Issue by ali-ramadhan** [climate-machine/Oceananigans.jl#23]
It'll be much cleaner. Not relevant until #13 is resolved.

**Comment by ali-ramadhan** [comment]
@peterahrens also suggested that we really wanna swizzle. We can transpose yC and reshape Nz to (1, 1, Nz) but ideally reshape all of them so the intent of the code is clear. We can also construct a ProductArray or ArrayOfBlockAxes.

```julia
"""
    ArrayOfBlockAxes

A convenient lazy array type which turns a tuple of blocked axes into an array
of axes for each block.
"""
struct ArrayOfBlockAxes{T, N, P <: NTuple{N, AbstractVector}} <: AbstractArray{T, N}
    parts::P
    function ArrayOfBlockAxes{T, N, P}(parts::P) where {T, N, P}
        @assert T <: Tuple{map(eltype, parts)...}
        return new{T, N, P}(parts)
    end
end

ArrayOfBlockAxes(parts) = ArrayOfBlockAxes{Tuple{map(eltype, parts)...}}(parts)
ArrayOfBlockAxes{T}(parts) where {T} = ArrayOfBlockAxes{T, length(parts), typeof(parts)}(parts)

Base.IndexStyle(::ArrayOfBlockAxes) = IndexCartesian()
Base.size(arr::ArrayOfBlockAxes) = map(part -> size(part, 1), arr.parts)
Base.size(arr::ArrayOfBlockAxes, d::Int) = size(arr.parts[d], 1)
Base.axes(arr::ArrayOfBlockAxes) = map(part -> axes(part, 1), arr.parts)
Base.axes(arr::ArrayOfBlockAxes, d::Int) = axes(arr.parts[d], 1)
(Base.getindex(arr::ArrayOfBlockAxes{T, N}, i::Vararg{<:Any, N})::T) where {T, N} = map(getindex, arr.parts, i)
```

**Comment by ali-ramadhan** [comment]
Oh nice this ancient issue is now resolved thanks to PR #343.
